pkg/labels/selector.go

```diff
@@ -800,6 +801,14 @@ func SelectorFromSetParse(ls Set) (Selector, error) {
 // ParseSelector takes a string representing a selector and returns an
 // object suitable for matching, or an error.
 func ParseSelector(selector string) (Selector, error) {
+	return parseVersionedSelector(selector, "", "")
+}
+
+func ParseVersionedFieldSelector(selector, version, apiResource string) (Selector, error) {
+	return parseVersionedSelector(selector, version, apiResource)
+}
+
+func parseVersionedSelector(selector, version, apiResource string) (Selector, error) {
```

Comment:
How about this for a compromise.

Change this to be "parseWithFilter", and pass in a function like ```func (key, value string) (newKey, newValue string, err error)```; then call this function instead of convertToInternalVersion.

That way we can keep this api conversion specific code out of the selector package. Does that make sense?